native/sorted_set_nif/src/supported_term.rs

```diff
@@ -133,6 +133,7 @@ impl PartialEq for SupportedTerm {
                         if self_inner[idx] != inner[idx] {
                             return false;
                         }
+                        idx += 1;
```

Comment:
Unless I'm missing something...this looks like an endless loop here without this change?